### DISCUSSION TITLE
Erros in console (tested on MacOS, Win10 x64) `Uncaught SyntaxError: Failed to construct 'WebSocket': The URL 'ws://localhost:undefined?connectToken=undefined' is invalid.`

### DISCUSSION THREAD

**Discussion by lukasMega** [neutralinojs/neutralinojs#1319] [Q&A]
Hello,
thank you for this awesome project! It deserves a broader audience and more attention.

____

UPDATE: 

no more relevant:

<details><summary>click to show old text</summary>

I have tried to follow official docs and on my Windows 10 (x64) PC there are problems.

My steps to reproduce:

1. follow guide: https://neutralino.js.org/docs/getting-started/using-frontend-libraries#react
 -  ```
    neu create myapp --template codezri/neutralinojs-react
    cd myapp
    
    # Start the React development server with Neutralinojs
    neu run
    ```
2. nothing is rendered

on my MacBook, it's working as expected.

I would be grateful to have more examples and guides, especially for TypeScript projects (React + Vite). 

</details>

---

Also I noticed some errors in console on MacOS (in `browser` mode) (`npx neu run -- --mode=browser`):

```
__neutralino_globals.js:1  Uncaught SyntaxError: Unexpected token '<' (at __neutralino_globals.js:1:1)
```

```
Uncaught SyntaxError: Failed to construct 'WebSocket': The URL 'ws://localhost:undefined?connectToken=undefined' is invalid.
    at l (index.js:1:1)
    at __webpack_modules__../node_modules/@neutralinojs/lib/dist/index.js.exports.init (index.js:1:1)
    at ./src/index.js (index.js:24:1)
    at options.factory (react refresh:6:1)
    at __webpack_require__ (bootstrap:24:1)
    at startup:7:1
    at startup:7:1
```

 

<img width="816" alt="Screen 2024-09-26 at 12 13 58" src="https://github.com/user-attachments/assets/74d713bd-42cd-4da3-9abc-8ada4fa3b8b3">

 - are these logs expected? There are appearing like every second


---

### window mode: 

<img width="1364" alt="Screen 2024-09-26 at 12 15 06" src="https://github.com/user-attachments/assets/9ac57d1e-c1dd-46a6-9e82-6697828114e6">



### browser mode: 

<img width="1668" alt="Screen 2024-09-26 at 12 13 36" src="https://github.com/user-attachments/assets/b5c73fa7-f9f8-4051-9721-87453e618295">




I will provide more details later today.

**Comment by lukasMega** [comment]
update, here are more details about my system (Windows 10 x64):


<details><summary>click to see output of <code>systeminfo</code> </summary>

```console
C:\Users\lm>systeminfo

Host Name:                 LM-MAXWELL
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.19045 N/A Build 19045
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          lm
Registered Organization:
Product ID:                00330-53913-19842-AAOEM
Original Install Date:     08/10/2023, 22:07:43
System Boot Time:          26/09/2024, 17:56:28
System Manufacturer:       Gigabyte Technology Co., Ltd.
System Model:              B550I AORUS PRO AX
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: AMD64 Family 25 Model 80 Stepping 0 AuthenticAMD ~3847 Mhz
BIOS Version:              American Megatrends International, LLC. FB, 08/06/2023
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume2
System Locale:             cs;Čeština
Input Locale:              en-us;Angličtina (Spojené státy)
Time Zone:                 (UTC+01:00) Praha, Bratislava, Budapešť, Bělehrad, Lublaň
Total Physical Memory:     32,122 MB
Available Physical Memory: 18,245 MB
Virtual Memory: Max Size:  34,170 MB
Virtual Memory: Available: 17,921 MB
Virtual Memory: In Use:    16,249 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    DNX-GROUP
Logon Server:              \\LM-MAXWELL
Hotfix(s):                 18 Hotfix(s) Installed.
                           [01]: KB5042097
                           [02]: KB5031988
                           [03]: KB5011048
                           [04]: KB5011052
                           [05]: KB5015684
                           [06]: KB5043131
                           [07]: KB5014032
                           [08]: KB5025315
                           [09]: KB5032392
                           [10]: KB5032907
                           [11]: KB5034224
                           [12]: KB5036447
                           [13]: KB5037240
                           [14]: KB5037995
                           [15]: KB5039336
                           [16]: KB5043935
                           [17]: KB5043130
                           [18]: KB5034441
Network Card(s):           2 NIC(s) Installed.
                           [01]: Realtek Gaming 2.5GbE Family Controller
                                 Connection Name: Ethernet
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.88.1
                                 IP address(es)
                                 [01]: 192.168.87.209
                                 [02]: fe80::1f26:1a9b:58c:7ab5
                           [02]: Realtek 8852CE WiFi 6E PCI-E NIC
                                 Connection Name: Wi-Fi
                                 Status:          Media disconnected
Hyper-V Requirements:      VM Monitor Mode Extensions: Yes
                           Virtualization Enabled In Firmware: Yes
                           Second Level Address Translation: Yes
                           Data Execution Prevention Available: Yes

C:\Users\lm>
```
</details>

Edge version: 
```
Microsoft Edge Version 92.0.902.67 (Official build) (64bit)
```

node.js and npm version:
```console
$ node -v && npm -v
v20.17.0
10.8.2
```

## Neu version:
```console
$ neu version
  _   _            _             _ _             _
 | \ | | ___ _   _| |_ _ __ __ _| (_)_ __   ___ (_)___
 |  \| |/ _ \ | | | __| '__/ _' | | | '_ \ / _ \| / __|
 | |\  |  __/ |_| | |_| | | (_| | | | | | | (_) | \__ \
 |_| \_|\___|\__,_|\__|_|  \__,_|_|_|_| |_|\___// |___/
                                               |__/
--- Global ---
neu CLI: v11.3.0 (latest)
neu: INFO Run this command inside your project directory to get project specific Neutralinojs version.
```

![msedge_BnTZTuXHC7](https://github.com/user-attachments/assets/c8b20623-b2de-44d6-b71e-143220bcfe26)


on my Windows PC, when `mode=chrome` is used the content is rendered (I hope it's rendered correctly, there are the same errors in the console like I noticed on MacOS)

![mode-chrome](https://github.com/user-attachments/assets/8d408956-6813-4a6f-8142-6f75245437f0)


---

# UPDATE: 

oh, I had very old version of Edge browser (because I'm using Firefox primarly)
 Also I had blocked network access to Edge browser in firewall (simplewall) - **so that was the reason I saw empty page with nothing rendered.** ❗ 
 - sorry for my local misconfiguration 

And also my PiHole was blocking bunch of micro$oft domains.
I temporarily allowed them, then I was able to update Edge to newest version (from ` 92.0.902.67` to `129.0.2792.52`)
But still the same errors in console


![scr_2024-09-26_20-47-28_Code](https://github.com/user-attachments/assets/3622b4d2-e51a-4080-a728-6a91f5c2d0ac)

**Comment by lukasMega** [comment]
I already wasted my time here, more than I wish.

I have tried to follow steps here https://neutralino.js.org/docs/getting-started/using-frontend-libraries but I have still those errors:

> [!caution]
> ## `Uncaught SyntaxError: Failed to construct 'WebSocket': The URL 'ws://localhost:undefined?connectToken=undefined' is invalid.`

I'm leaving this project. Sorry. I hope it will be better, I wish all good.
Maybe it is something wrong on my both environments (Windows 10 and MacOS) 🤷‍♂️ ... but I've successfully switched to another alternative which is working well for me (however it's not as simple as Neutralino; and no, it's not Electron).

**Comment by metacurb** [comment]
I am having the inverse of this error. I have a project set up which works fine on Windows. However, Ive tried to set it up on my mac and I get the above error. I tried to configure a new project and end up with the same result.

**Comment by mathiasayivor** [comment]
Checkout [modes.window.injectGlobals](https://neutralino.js.org/docs/configuration/neutralino.config.json#modeswindowinjectglobals-boolean) in your `neutralino.config.json`
